lazyServer.go

```diff
@@ -0,0 +1,33 @@
+package multistream
+
+import (
+	"io"
+	"sync"
+)
+
+type lazyServerConn struct {
```

Comment:
can we get a doc comment saying what the purpose of `lazyServerConn` is exactly?